core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -0,0 +1,746 @@
+
+
+  final def claimAndConfigureGoogleProject: MigrateAction[Unit] =
+    for {
+      migration <- findMigration { m =>
+        m.started.isDefined && m.newGoogleProjectConfigured.isEmpty
+      }
+
+      workspace <- getWorkspace(migration.workspaceId)
+      errorDetails = Map(
+        ("migrationId" -> migration.id),
+        ("workspaceId" -> workspace.workspaceId)
+      )
+
+      _ <- MigrateAction.raiseWhen(workspace.billingAccountErrorMessage.isDefined) {
+        WorkspaceMigrationException(
+          message = "Workspace migration failed: billing account error exists on workspace",
+          data = errorDetails + ("billingAccountErrorMessage" -> workspace.billingAccountErrorMessage.get)
+        )
+      }
+
+      workspaceBillingAccount <- MigrateAction.liftIO(IO {
+        workspace.currentBillingAccountOnGoogleProject.getOrElse(
+          throw WorkspaceMigrationException(
+            message = "Workspace migration failed: no billing account on workspace",
+            data = errorDetails
+          )
+        )
+      })
+
+      // Safe to assume that this exists if the workspace exists
+      billingProject <- inTransactionT { _
+        .rawlsBillingProjectQuery
+        .load(RawlsBillingProjectName(workspace.namespace))
+      }
+
+      billingProjectBillingAccount <- MigrateAction.liftIO(IO {
+        billingProject.billingAccount.getOrElse(
+          throw WorkspaceMigrationException(
+            message = "Workspace migration failed: no billing account on billing project",
+            data = errorDetails + ("billingProject" -> billingProject.projectName)
+          )
+        )
+      })
+
+      _ <- MigrateAction.raiseWhen(billingProject.invalidBillingAccount) {
+        WorkspaceMigrationException(
+          message = "The migration failed: invalid billing account on billing project",
+          data = errorDetails ++ Map(
+            ("billingProject" -> billingProject.projectName),
+            ("billingProjectBillingAccount" -> billingProjectBillingAccount)
+          )
+        )
+      }
+
+      _ <- MigrateAction.raiseWhen(workspaceBillingAccount != billingProjectBillingAccount) {
+        WorkspaceMigrationException(
+          message = "Workspace migration failed: billing account on workspace differs from billing account on billing project",
+          data = errorDetails ++ Map(
+            ("workspaceBillingAccount" -> workspaceBillingAccount),
+            ("billingProject" -> billingProject.projectName),
+            ("billingProjectBillingAccount" -> billingProjectBillingAccount)
+          )
+        )
+      }
+
+      workspaceService <- MigrateAction.asks(_.workspaceService)
+      (googleProjectId, googleProjectNumber) <- MigrateAction.liftIO {
+        workspaceService.setupGoogleProject(
+          billingProject,
+          workspaceBillingAccount,
+          workspace.workspaceId,
+          workspace.toWorkspaceName
+        ).io
+      }
+
+      configured <- nowTimestamp
```

Comment:
pretty minor nitpick: feels like this could be named something more like "now" or "timestamp" and that might be a bit clearer below.